Tests/LiquidProjections.NHibernate.Specs/NHibernateProjectorSpecs.cs

```diff
@@ -1289,6 +1302,110 @@ public void Then_it_should_use_the_header()
+                        {
+                            if (p.Category == @event.Category)
+                            {
+                                throw new InvalidOperationException("The event should not be handled twice");
+                            }
+
+                            p.Category = @event.Category;
+                            p.Name = @event.Name;
+                        });
+
+                    Transaction transaction = new TransactionBuilder()
+                        .WithCheckpointNumber(5)
+                        .WithEvent(new ProductAddedToCatalogEvent
+                        {
+                            Category = "some category",
+                            Name = $"some product",
+                            ProductKey = $"some key",
+                        })
+                        .Build();
+
+                    transactions = new List<Transaction> {transaction}.AsReadOnly();
+
+                    info = new SubscriptionInfo();
+
+                    StartProjecting();
+                });
+
+                When(async () =>
+                {
+                    await Subject.Handle(transactions, info);
+                    await Subject.Handle(transactions, info);
+                });
+            }
+
+            [Fact]
+            public void Then_it_should_not_handle_the_transactions()
+            {
+                ProjectionException.Should().BeNull();
+            }
+        }
+        
+        #endregion
+        
+        #region Filtering
+        
+        public class When_the_projection_being_updated_doesnt_pass_the_filter :
+            Given_a_sqlite_projector_with_an_in_memory_event_source
+        {
+            public When_the_projection_being_updated_doesnt_pass_the_filter()
+            {
+                Given(() =>
+                {
+                    Events.Map<ProductMovedToCatalogEvent>()
+                        .AsUpdateOf(@event => @event.ProductKey)
+                        .Using((product, @event, _) => product.Category = @event.Category);
+
+                    using (ISession session = The<ISessionFactory>().OpenSession())
+                    {
+                        session.Save(new ProductCatalogEntry
+                        {
+                            Id = "c350E",
+                            Category = "Old Value"
+                        });
+                        
+                        session.Flush();
+                    }
+
+                    //Subject.Filter = projection => !projection.Corrupted; 
+                    
+                    StartProjecting();
+                });
+
+                When(() => The<MemoryEventSource>().Write(new ProductAddedToCatalogEvent
+                {
+                    ProductKey = "c350E",
+                    Category = "New Value"
+                }));
+            }
+
+            [Fact]
+            public void Then_it_should_not_update_the_projection()

```

Comment:
Isn't the default behavior that it will update the projection if you don't provide the filter?
`public Predicate<TProjection> Filter { get; set; } = _ => true;`

Reply:
You're right, but the spec wasn't doing it was supposed to do.